doc/man7/EVP_KDF_SS.pod

```diff
@@ -0,0 +1,224 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF_SS - The Single Step EVP_KDF implementation. (Also known as One
+-Step Key Derivation).
+
+=head1 DESCRIPTION
+
+Support for computing the B<SSKDF> KDF through the B<EVP_KDF> API.
+The EVP_KDF_SS algorithm implements the Single Step KDF key derivation function.
+SSKDF derives a key using input such as a shared secret key (that was generated
+during the execution of a key establishment scheme) and fixedinfo.
+
+=head2 Auxilary function
+
+The implementation uses a selectable auxiliary function H, which can be one of:
+
+=over 4
+
+=item B<H(x) = hash(x, digest=md)>
+
+=item B<H(x) = HMAC_hash(x, key=salt, digest=md)>
+
+=item B<H(x) = KMACxxx(x, key=salt, custom="KDF", outlen=mac_size)>
+
+=back
+
+Both the HMAC and KMAC implementations set the key using the 'salt' value.
+The hash and HMAC also require the digest to be set.
+
+=head2 Numeric identity
+
+B<EVP_KDF_SS> is the numeric identity for this implementation; it
+can be used with the EVP_KDF_CTX_new_id() function.
+
+=head2 Supported controls
+
+The supported controls are:
+
+=over 4
+
+=item B<EVP_KDF_CTRL_SET_MD>
+
+=item B<EVP_KDF_CTRL_SET_MAC>
+
+=item B<EVP_MAC_CTRL_SET_MAC_SIZE>
+
+=item B<EVP_KDF_CTRL_SET_SALT>
+
+These controls work as described in L<EVP_KDF_CTX(3)/CONTROLS>.
+
+=item B<EVP_KDF_CTRL_SET_SSKDF_SECRET>
+
+This control expects two arguments: C<unsigned char *secret>, C<size_t secretlen>
+
+The shared secret used for key derivation.  This control sets the secret.
```

Comment:
Isn't this the same as `EVP_KDF_CTRL_SET_KEY`?

Reply:
I would prefer to leave it as secret, since the standard refers to it as the 'shared secret'
(similiar to EVP_KDF_CTRL_SET_SSKDF_SECRET).

Reply:
Hmmmm...but it's essentially the same thing as `EVP_KDF_CTRL_SET_KEY` right? I understand trying to keep things consistent with the spec...but consistency with other kdfs is also important.

Reply:
Okie, so I can see that a name that relates to the standard is a good thing.  I disagree with the need to make it SSKDF specific, though, so as a middle ground, I would suggest making `EVP_KDF_CTRL_SET_SECRET` an alias for `EVP_KDF_CTRL_SET_KEY`.

Reply:
(and yeah, there is a `EVP_KDF_CTRL_SET_TLS_SECRET `...  I cannot recall what made TLS special enough to warrant its very own mnemonic)

Reply:
Maybe I will just change it to set_key and explain it in the doc for single step. I think it does get confucing also using the alias, EVP_KDF_CTRL_SET_TLS_SECRET doesnt help things either..

Reply:
I may keep the ctrl string names as "secret" & "hexsecret"

Reply:
Also handles aliases for "key" & "hexkey"